### ISSUE TITLE
CPU usage even when no file loaded

### ISSUE THREAD

**Issue by dw817** [SubtitleEdit/subtitleedit#4384]
I'm concerned that even when no SRT or other subtitle file is loaded the executable can use anywhere from 5-10% CPU doing ... absolutely nothing. If I leave it for hours, it jumps up to 15%.

What could possibly be happening in the executable when there is no file even loaded in it ?

**Comment by OmrSi** [comment]
This is not happening with me:
![image](https://user-images.githubusercontent.com/20923700/93915468-c256e480-fd10-11ea-98e3-6c9f2ee10ddd.png)

**Comment by niksedk** [comment]
Also zero cpu% here:
![image](https://user-images.githubusercontent.com/1261331/93915759-c121b980-fd08-11ea-9700-2ec3f567cb98.png)

**Comment by dw817** [comment]
Glad it's not happening with you, my question still remains. Or let's put it this way. What could CAUSE that to happen as it is here ?
I just brought it back up NOW with no file and it's using 5.29% CPU.
I'd like to hear from the people that DO have this problem as it may be a rare or isolated incident and what they think might be causing it.
Simply stating "Not me" doesn't answer this question.

**Comment by dw817** [comment]
![subtitleedit cpu](https://user-images.githubusercontent.com/4763989/93943384-08ec1500-fcf8-11ea-879a-9484d5c798ea.png)

**Comment by dw817** [comment]
![09-22-20 System Info](https://user-images.githubusercontent.com/4763989/93944452-6ed99c00-fcfa-11ea-821a-2c0241315dfc.jpg)

**Comment by dw817** [comment]
![09-22-20 System Info](https://user-images.githubusercontent.com/4763989/93944574-b95b1880-fcfa-11ea-8939-88fc4cb16084.jpg)

**Comment by OmrSi** [comment]
Simply stating "Not me" is a first step in the course of identifying the cause of the issue.

**Comment by dw817** [comment]
Reminds of a bunch of kids, really. Very well, OmrSi, do you have anything more concrete to to contribute towards the correct answer to my initial question ?

I've posted my system status. Not sure what other data in my computer would be deemed helpful to solving this question.

**Comment by vinodthebond** [comment]
@dw817 , please be respectful to the hard work these guys are putting in.  This piece of software is by far the best free subtitling tool available out there.   There could be some memory leakage happening.  It would certainly take some time to look into this and find out where the leakage is happening.

I know, "Not me" is not an answer.  But when the lead developer(@niksedk)  says something, we have to respect.

I suppose, closing the application itself would help you when it is not in use.

Let's put all this aside.  I'm sure @niksedk  will have a resolution for the same.

**Comment by dw817** [comment]
I see ... I certainly respect niksedk for being the developer. I would like to mention it is a key program I use to analyze and repair misaligned and offtrack subtitles. Its ability to convert to another language has also been quite instrumental in converting English subtitles to German for my German speaking friend and then hardcoding them as new DVDs.

As you know or should Google is now actually charging money if you try to convert text past I believe it's 8000 characters to another language. SubtitleEdit has blissfully been able to do this for SRTs cost free. So there is a lot of good in SubtitleEdit. I place its value right there with VirtualDUB as an absolutely essential tool for video editing.

So - great respect for the program itself and its authors. I did not have this CPU leakage problem appear in Windows 8 so that might help in the investigation. Windows 10 has always given me problems and it was not my CPU choice at time of purchase. :) Also the version of SE I am using is 3516.

The program that announced this CPU leakage is Process Explorer v 16.22 by Mark Russinovich. I have other tools I've written that run in the background at 0% CPU usage so it is a possible number to reach.

Thank you for your prompt and compassionate reply, vinodthebond.

My hope is the problem can be resolved outside of shutting it down when not in use - which obviously removes the CPU usage.

**Comment by vinodthebond** [comment]
Thank you for the kind words, @dw817.  Now that you have more provided information, this could help the dev to narrow down the cause of the issue.  The issue might have also been caused due to the modification to Windows by your PC manufacturer.  Let's wait for the dev to reply.

**Comment by niksedk** [comment]
@dw817: Could you see if this beta is any better: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.16/SubtitleEditBeta.zip ?

Does anybody else have this problem?

**Comment by dw817** [comment]
Well done, niksedk !

That fixed it. I should point out your BETA loads quite a bit faster too. On my computer the original 3516 takes about 3-seconds before the initial frame appears. Executing your BETA it comes up instantly.

What did you change ?

**Comment by niksedk** [comment]
The two main timers now exists early if no sub is loaded: Code change here: https://github.com/SubtitleEdit/subtitleedit/commit/fcc81913fce53d825cadd3bdafdda9aac6cd0150

**Comment by dw817** [comment]
That got it ! Now when I load a subtitle and leave it in the background though the CPU still gets used. Is there a way to maintain the frictionless CPU usage even when a subtitle is loaded ?

In any case, makes a big difference here what you did. Now as I've learned in coding once you correct something somewhere for someone it usually breaks for someone else. :) Hopefully this won't happen here. Thank you so much for taking the time to correct this so it at least the stray CPU usage doesn't happen on an empty file.

Closing this ticket out. Good job, niksedk !
